core/src/main/scala/kafka/server/AbstractAsyncFetcher.scala

```diff
@@ -31,19 +36,50 @@ import org.apache.kafka.common.requests.EpochEndOffset._
 import org.apache.kafka.common.requests.{EpochEndOffset, FetchRequest, FetchResponse, OffsetsForLeaderEpochRequest}
 import org.apache.kafka.common.{InvalidRecordException, TopicPartition}
 
-import java.nio.ByteBuffer
-import java.util.Optional
-import java.util.concurrent.TimeUnit
-import java.util.function.Consumer
 import scala.collection.JavaConverters._
 import scala.collection.{Map, Seq, Set, mutable}
 import scala.math.min
 
 
+sealed trait FetcherEvent extends Comparable[FetcherEvent] {
+  def priority: Int // an event with a higher priority value is more important
```

Comment:
do we capture ordering across fetcher event? 
Is there a case (or future use case) that a higher priority event would have to depend on a lower priority event?
If so, do we want to introduce a dependency graph for these events? 
Or this priority only makes sense for different partitions, if so, could you document it somewhere.

Reply:
I haven't found the need for dependencies between the FetcherEvents. That is, the current event types are all conceptually independent of each other: AddPartitions, RemovePartitions, GetPartitionCount, and TruncateAndFetch.
And I don't expect to introduce dependencies in the future.

Not sure I entirely get your concern. If you think there is still confusion, please give an example.

Reply:
discussed offline, if there is a ordering requirement,  the caller whoever enqueue the event will ensure the ordering e.g., remove parition, wait for completion, and then add partition, etc.